Python/pystate.c

```diff
@@ -1426,7 +1426,12 @@ _str_shared(PyObject *obj, _PyCrossInterpreterData *data)
 static PyObject *
 _new_long_object(_PyCrossInterpreterData *data)
 {
+#if ( ULONG_MAX == UINT_MAX )
+    return PyLong_FromLongLong((int32_t)(data->data));
```

Comment:
```suggestion
    return PyLong_FromLongLong((intptr_t)(data->data));
```

With the above change, there is no need for the `#if` check at all, since `intptr_t` is signed and should be the same size as `void*` on all platforms.

Reply:
nice find. Thx!